Source/WebCore/platform/graphics/filters/FilterOperations.h

```diff
@@ -37,20 +38,37 @@ struct BlendingContext;
 class FilterOperations {
     WTF_MAKE_FAST_ALLOCATED;
 public:
+    using const_iterator = Vector<Ref<FilterOperation>>::const_iterator;
+    using const_reverse_iterator = Vector<Ref<FilterOperation>>::const_reverse_iterator;
+    using value_type = Vector<Ref<FilterOperation>>::value_type;
+
     FilterOperations() = default;
-    WEBCORE_EXPORT explicit FilterOperations(Vector<RefPtr<FilterOperation>>&&);
+    WEBCORE_EXPORT explicit FilterOperations(Vector<Ref<FilterOperation>>&&);
 
     WEBCORE_EXPORT bool operator==(const FilterOperations&) const;
 
-    void clear() { m_operations.clear(); }
+    FilterOperations copy() const
+    {
+        return FilterOperations { m_operations.map([](const auto& op) { return op.copyRef(); }) };
+    }
+
+    FilterOperations clone() const
+    {
+        return FilterOperations { m_operations.map([](const auto& op) { return op->clone(); }) };
+    }
```

Comment:
Having both a `copy` and `clone` is confusing. 
Do we really need both ? If yes, should we consider other names ? shallowCopy/deepCopy ?

Reply:
We can just remove copy() and make FilterOperations do the right thing under copy construction.